### PR TITLE
support producer-fast mode for plugin

### PR DESCRIPTION
**Pull request by vasvf**
These small changes add support for "producer-fast" mode in plugin.

There's also an addition (117-118 lines) where it removes "CC" override in case project was switched from consumer to producer

### CODE REVIEW

File: cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb [hunk 1]
```diff
@@ -169,7 +171,7 @@ def self.enable_xcremotecache(target, repo_distance, xc_location, xc_cc_path, mo
         postbuild_script.dependency_file = "$(TARGET_TEMP_DIR)/postbuild.d"
 
         # Mark a sha as ready for a given platform and configuration when building the final_target
-        if mode == 'producer' && target.name == final_target
+        if (mode == 'producer' || mode == 'producer') && target.name == final_target
```
Comment (polac24) [inline cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb:174]: probably a typo for `producer-fast`

**Review by polac24 — COMMENTED** [review]
Why do want to add support for producer-fast in a CocoaPods plugin?
This feature is veeery experimental and there is no active development nor plans to make it GA.

File: cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb (hunk 1 above)
Comment (vasvf) [reply cocoapods-plugin/lib/cocoapods-xcremotecache/command/hooks.rb]: for sure it is! :)

File: Sources/XCRemoteCache/Commands/Prepare/Integrate/BuildSettingsIntegrateAppender.swift [hunk 2]
```diff
@@ -47,6 +47,8 @@ class XcodeProjBuildSettingsIntegrateAppender: BuildSettingsIntegrateAppender {
             result["CC"] = wrappers.cc.path
             result["LD"] = wrappers.ld.path
             result["LIBTOOL"] = wrappers.libtool.path
+        } else {
+            result.removeValue(forKey: "CC")
```
[the remark this answers is not in the document]
Comment (vasvf) [reply Sources/XCRemoteCache/Commands/Prepare/Integrate/BuildSettingsIntegrateAppender.swift]: Hm. Did not mention class name. Removed it for now

**Review by polac24 — APPROVED** [review]

**Pull request merged**